Angular/src/utils/isOverlapRecurrentAppointment.ts

```diff
@@ -0,0 +1,118 @@
+import {AppointmentAddingEvent} from 'devextreme/ui/scheduler';
+import {RRule, rrulestr} from 'rrule';
+import {Appointment} from '../app/interfaces';
+
+const MS_IN_HOUR = 60000;
+
+function resetDateTime(date: Date): Date {
+  return new Date(date.getFullYear(), date.getMonth(), date.getDate());
+}
+function getAllDayStartTime(date: Date): number {
+  return resetDateTime(new Date(date.getTime() + date.getTimezoneOffset() * MS_IN_HOUR)).getTime();
+}
+function getAllDayEndTime(endTime: number): number {
+  let endDate = new Date(endTime);
+  endDate = new Date(endDate.getFullYear(), endDate.getMonth(), endDate.getDate() + 1);
+  return endDate.getTime();
+}
+
+function convertRecurrenceExceptions(dateStrings: string): Date[] {
+  return dateStrings.split(",").map(dateString => {
+    return new Date(dateString.slice(0, 4) + "-" + dateString.slice(4, 6) + "-" + dateString.slice(6, 8) + "T" + dateString.slice(9, 11) + ":" + dateString.slice(11, 13) + ":" + dateString.slice(13, 15) + "Z");
+  });
+}
```

Comment:
Minor suggestion.

I don't find any efficient way to parse this date format quickly.
Maybe we should write it with string template replacements?

```suggestion
function convertRecurrenceExceptions(dateStrings: string): Date[] {
  return dateStrings.split(",").map(dateString => {
    const formattedDateString = `${dateString.slice(0, 4)}-${dateString.slice(4, 6)}-${dateString.slice(6, 8)}T${dateString.slice(9, 11)}:${dateString.slice(11, 13)}:${dateString.slice(13, 15)}Z`;
    return new Date(formattedDateString);
  });
}
```

@andrewmakarov what do you think about it?
Maybe we should change the recurrent exception date format in the future?